packages/example-studio/components/FunkyEditor/FunkyEditor.js

```diff
@@ -0,0 +1,37 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import SlateEditor from '@sanity/form-builder/lib/inputs/BlockEditor-slate'
+export default class FunkyEditor extends React.Component {
+  static propTypes = {
+    type: PropTypes.shape({
+      title: PropTypes.string
+    }).isRequired,
+    level: PropTypes.number,
+    value: PropTypes.array,
+    onChange: PropTypes.func.isRequired
+  };
+
+  constructor(props) {
+    console.log(props)
+    super(props)
+  }
+
+  handleChange = event => {
+    this.props.onChange(event)
```

Comment:
If all handleChange do is just call the given `onChange`-prop, maybe just pass it from render()?
```js
render() {
  const {type, value, level, onChange} = this.props
  return (
    //...
    <SlateEditor
      //...
      onChange={onChange}
     />
    //...
  )
}
```